portal/models/user.py

```diff
@@ -1171,7 +1171,8 @@ def fuzzy_match(self, first_name, last_name, birthdate):
         # birthdate is trickier - raw delta doesn't make sense.  treat
         # it like a string, mismatch always results in a 0 score
         dob = self.birthdate or datetime.utcnow()
-        if (dob.strftime('%d%m%Y') != birthdate.strftime('%d%m%Y')):
+        if (birthdate.year < 1900 or
+            dob.strftime('%d%m%Y') != birthdate.strftime('%d%m%Y')):
```

Comment:
Right, with this change, it no longer raises an exception. Now the conditional is false before it even reaches the `.strftime()` call, so it just moves on to the `return 0` line (same as if the dates didn't match in the first place)